opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/__init__.py

```diff
@@ -196,24 +213,52 @@ def create_observable_counter(
             self._instrument_id_instrument[instrument_id] = instrument
             return instrument
 
-    def create_histogram(self, name, unit="", description="") -> APIHistogram:
+    def create_histogram(
+        self,
+        name: str,
+        unit: str = "",
+        description: str = "",
+        advisory: Optional[MetricsHistogramAdvisory] = None,
+    ) -> APIHistogram:
+        if advisory is not None:
+            invalid_advisory = False
+            try:
+                boundaries = advisory.explicit_bucket_boundaries
+                invalid_advisory = not (
+                    boundaries
+                    and all(isinstance(e, float) for e in boundaries)
```

Comment:
```suggestion
                    and all(isinstance(e, (float, int)) for e in boundaries)
```

consider allowing ints to be passed or just do a `float()` call as well. It's a bit confusing to not accept int because [PEP484 already works this way](https://peps.python.org/pep-0484/#the-numeric-tower) when typechecking against `float`.

Reply:
Readded